docs/source/tools/trigger-service.rst

```diff
@@ -0,0 +1,119 @@
+.. Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Trigger Service
+###############
+
+**WARNING:** The triggers-service is an early access feature that is actively being designed and is subject to breaking changes. The documentation here at this time is limited to basic usage. As more features become available the documentation will be updated to include them. We welcome feedback about the trigger-service on our issue tracker, our forum, or on Slack.
+
+The `DAML triggers <../triggers/index.html#running-a-daml-trigger>`_ documentation shows a simple method using the ``daml trigger`` command to arrange for the execution of a single trigger. Using this method, a dedicated process is launched to host the trigger.
+
+Complex workflows can require running many triggers for many parties and at a certain point, use of ``daml trigger`` with its process per trigger model becomes unwieldy. The trigger-service provides the means to host multiple triggers for multiple parties running against a common ledger in a single process and provides a convenient interface for starting, stopping and monitoring them.
+
+The trigger-service is a ledger client that acts as an end-user agent. The trigger service intermediates between the ledger and end-users by running triggers on their behalf. The trigger-service is a REST service that interacts with end-users by exchanging HTTP requests and responses. Request/response data exchanged between the end-users and the trigger-service is in the JSON data-interchange format.
+
+Starting the trigger-service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, it is assumed there is a sandbox ledger running on port 6865 on the local host.
+
+.. code-block:: bash
+
+   daml trigger-service --ledger-host localhost --ledger-port 6865 --wall-clock-time --no-secret-key
+
+The above starts the trigger-service using a number of default parameters. Most notably, the HTTP port the trigger service listens on which defaults to 8088. The meanining of the ``--no-secret-key`` parameter will be explained in the next section. To see all of the available parameters, their defaults and descriptions, one can execute the command ``daml trigger-service --help``.
+
+Although as we'll see, the trigger-service exposes an endpoint for end-users to upload DAR files to the service (DAML archives containing triggers), it is sometimes convenient to start the service preconfigured with a specific DAR. To do this, the ``--dar`` option is provided.
+
+.. code-block:: bash
+
+   daml trigger-service --ledger-host localhost --ledger-port 6865 --wall-clock-time --no-secret-key --dar <DAR>
+
+Identifying trigger parties
```

Comment:
Similar to my comment below, I would suggest to move the requests above this so people can get started quickly. If I see a `--user` option to `curl` I can probably guess what it does and if I cannot I can still read the details below.

Reply:
Ok.

Reply:
✔️